src/pages/index.js

```diff
@@ -0,0 +1,354 @@
+      likeButton.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
+function handleImageClick(data) {
+  previewModalImageElement.src = data.link;
+  previewModalImageElement.alt = data.name;
+  previewModalCaptionElement.textContent = data.name;
+  openModal(previewModal);
+}
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardNameElement = cardElement.querySelector(".card__info");
+  const cardImageElement = cardElement.querySelector(".card__photo");
+  const cardLikeBtn = cardElement.querySelector(".card__like-button");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-button");
+
+  //TODO - if the card is liked, set the active class on the card
+  console.log(data);
+  if (data.isLiked) cardLikeBtn.classList.add("card__like-button_liked");
+
+  cardNameElement.textContent = data.name;
+  cardImageElement.src = data.link;
+  cardImageElement.alt = data.name;
+
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+  cardDeleteBtn.addEventListener("click", () => {
+    handleDeleteCard(cardElement, data._id);
+  });
+  cardImageElement.addEventListener("click", () => handleImageClick(data));
+
+  return cardElement;
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalEsc);
+  modal.addEventListener("click", handleOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeModalEsc);
+  modal.removeEventListener("click", handleOverlay);
+}
+
+function closeModalEsc(evt) {
+  if (evt.key === "Escape") {
+    const modalOpen = document.querySelector(".modal_opened");
+    closeModal(modalOpen);
+  }
+}
+
+function handleOverlay(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function handleProfileFormSubmit(evt) {
+  evt.preventDefault();
+  //helpers
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // helpers
+      setButtonText(submitBtn, false);
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  //helpers
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  const inputValues = {
+    name: addModalCaptionInput.value,
+    link: addModalLinkInput.value,
+  };
+
+  api
+    .postNewCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      evt.target.reset();
+      // disableButton(cardSubmitBtn, settings);
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute.


Please, uncomment it

You should **export** the `disableButton` function from `validation.js`, to be able to use it in `index.js`